open-api/rest-catalog-open-api.yaml

```diff
@@ -2893,6 +3003,37 @@ components:
           additionalProperties:
             type: string
 
+    AppendDataFileRequest:
+      type: object
+      required:
+        - data-files
+      properties:
+        identifier:
+          allOf:
+            - $ref: '#/components/schemas/TableIdentifier'
+          description: Table identifier to update; must be present for CommitTransactionRequest
+        accept-delay-ms:
+          type: integer
+          format: int64
+          description: Acceptable delay in milliseconds for processing the append request.
+          example: 300000
+        data-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/DataFile'
+          description: A list of serialized data files to append to the table metadata.
```

Comment:
shall we add a limit to number of data files to be appended in a single operation?

Reply:
Not sure what we mean by limit but I think it makes sense to define the expected HTTP error code in case the service cannot handle the list of data files.